xcode/Subconscious/Shared/Library/URLUtilities.swift

```diff
@@ -8,6 +8,37 @@
 import Foundation
 
 extension URL {
+    /// Attempt to construct a URL from the passed string and validate its components
+    /// This will only allow http & https URLs with a full hostname
+    init?(validatedString string: String) {
+        guard let url = URL(string: string) else {
+            return nil
+        }
+        
+        var components = URLComponents()
+        components.scheme = url.scheme
+        components.host = url.host
+        components.path = url.path
+        
+        guard let scheme = components.scheme,
+              ["http", "https"].contains(scheme.lowercased()) else {
+            return nil
+        }
+        guard components.host != nil else {
+            return nil
+        }
+        
+        if components.path.isEmpty {
+            components.path = "/"
+        }
+        
+        guard let validatedUrl = components.url else {
```

Comment:
Because `components` is only assigned scheme, host, and path, we will lose other parts of the URL, including fragment, query, etc.